src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/dynamicProgramming/DynamicProgramming.java

```diff
@@ -0,0 +1,74 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.dynamicProgramming;
+
+import se.liu.ida.hefquin.engine.federation.FederationAccessException;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpJoin;
+import se.liu.ida.hefquin.engine.queryplan.physical.BinaryPhysicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.*;
+import se.liu.ida.hefquin.engine.queryproc.*;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.utils.CardinalityEstimation;
+
+import java.util.*;
+
+public class DynamicProgramming extends CardinalityEstimation {
+    protected final QueryProcContext ctxt;
+    protected List<PhysicalPlan> lpList;
+
+    public DynamicProgramming(final QueryProcContext ctxt, final List<PhysicalPlan> lpList) {
+        assert ctxt != null;
+        this.ctxt = ctxt;
+        this.lpList = lpList;
+    }
+
+    protected PhysicalPlan optimizePhysicalPlanForMultiwayJoin() throws FederationAccessException {
```

Comment:
Shouldn't this method be public, given that this is the actual main method you are calling when using this class. 

Also, the methods here should not throw a FederationAccessException but a QueryOptimizationException.